benchmarks/tensorexpr/microbenchmarks.py

```diff
@@ -0,0 +1,236 @@
+
+for name, lmbda, pt_fn in custom_ops:
+    names.append(name)
+    nnc_fns.append(gen_custom_nnc_fun(lmbda))
+    pt_fns.append(gen_custom_torch_fun(pt_fn))
+
+benchmarks = list(zip(names, nnc_fns, pt_fns))
+
+def run_benchmarks(benchmarks):
+    df = pd.DataFrame(columns=['name', 'N', 'M', 'nnc_time', 'torch_time', 'ratio'])
+    sizes = [1, 4, 16, 64, 256, 1024]
+    with torch.no_grad():
+        for name, nnc_fun, torch_fun in benchmarks:
+            for N in sizes:
+                for M in sizes:
+                    iters = int(1e5 / (N + M))
+                    with kernel_arena_scope():
+                        dtype = torch._C.te.Dtype.Float
+
+                        dM = torch._C.te.ExprHandle.int(M)
+                        dN = torch._C.te.ExprHandle.int(N)
+
+                        dims_MN = [dM, dN]
+                        A = torch._C.te.Placeholder('A', dtype, [dM, dN])
+                        B = torch._C.te.Placeholder('B', dtype, [dM, dN])
+
+
+                        dim_args = [torch._C.te.DimArg(*args) for args in [(dM, 'm'), (dN, 'n')]]
+
+                        compute = nnc_fun(A, B)
+                        X = torch._C.te.Compute('X', dim_args, compute)
+                        loopnest = torch._C.te.LoopNest([X])
+                        loopnest.prepare_for_codegen()
+                        stmt = torch._C.te.simplify(loopnest.root_stmt())
+                        cg = torch._C.te.construct_codegen('llvm', stmt, [torch._C.te.BufferArg(x) for x in [A, B, X]])
+
+                        tA = torch.rand(M, N).clamp(0.01, 0.99)
+                        tB = torch.rand(M, N).clamp(0.01, 0.99)
+                        tX = torch.empty(M, N)
+                        tR = torch.empty(M, N)
+
+
+                        # warmup
+                        for _ in range(10):
+                            cg.call([tA, tB, tX])
+                        start = time.time()
+                        for it in range(iters):
+                            cg.call([tA, tB, tX])
+                        time1 = time.time() - start
+
+
+                        fn = torch_fun(tA, tB, tR)
+                        # warmup
+                        for _ in range(10):
+                            fn()
+                        start = time.time()
+                        for it in range(iters):
+                            fn()
+                        time2 = time.time() - start
+
+
+                        df = df.append({'name': name, 'N': N, 'M': M, 'nnc_time': time1, 'torch_time': time2, 'ratio': time2 / time1}, ignore_index=True)
+                        print(name, N, M)
+                        print(time2 / time1, time1, time2)
+                        print()
+
+                        def check_correctness(a, b):
+                            diff = abs(a.sum() - b.sum())
+                            if not (diff < 2.0):
+                                if a.bool().sum() == b.bool().sum():
```

Comment:
What does this do?

Reply:
Whups, this was a check I added specifically to make sure the boolean comparators were handled correctly. This is no longer right...